src/components/DefaultImage.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+function DefaultImage({idx, onClick, src, srcSet, sizes, height, width, alt, margin}){
+	style.margin = margin;
+	return (
+		<div style={style}> 
+			<a href="#" onClick={(e) => onClick(idx, e)}>
+				<img src={src} srcSet={srcSet} sizes={sizes} style={{display:'block', border:0}} height={height} width={width} alt={alt} />
```

Comment:
Move the `style` object out of the component. Provide style as reference. Now a new object is created on every render, resulting in unnecessary re-renders. Like you're already doing with the div style.

```js
const imageStyle = { display: 'block', border: 0 };

function DefaultImage() { //...
   return <image style={imageStyle} />
```